src/services/array-functions.js

```diff
@@ -1,39 +1,60 @@
 
 //in the function map, create a new array and store in a variable
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop into the call to fnc
 //add the returned value from fnc to the new array
 //return the new array
 export function map(theArray, fnc){
-
+  let myArray = [];
+  for(let i = 0; i < theArray.length; i++){
+    let athing = fnc(theArray[i]);
+    myArray.push(athing);
+  }
+  return myArray;
 }
 
+
+//newNames array is being returned correctly by console.log()-
+//but it is throwing undefined error when running test...
+//not sure why.
+
+
+
+
 //create a new array
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
+
+
 export function filter(theArray, fnc){
+  const newArray = theArray.filter(fnc);
```

Comment:
Make sure you aren't using the built in filter function to build your filter function. We need to understand how the filter works under the hood